ArmPkg/Drivers/ArmGicDxe/GicV3/ArmGicV3Dxe.c

```diff
@@ -628,7 +628,7 @@ GicV3DxeInitialize (
 
   RegValue = ArmGicV3GetControlSystemRegisterEnable ();
   if ((RegValue & ICC_SRE_EL2_SRE) == 0) {
-    ArmGicV3SetControlSystemRegisterEnable (RegValue | ICC_SRE_EL2_SRE);
+    ArmGicV3SetControlSystemRegisterEnable ((UINT32)RegValue | ICC_SRE_EL2_SRE);

```

Comment:
Everywhere `RegValue` is used a `UINT32` is required. Instead of the casts, can we just change `RegValue` to be a `UINT32`?